specification/trace/semantic_conventions/instrumentation/elasticsearch.md

```diff
@@ -0,0 +1,44 @@
+# Semantic conventions for Elasticsearch
+
+**Status**: [Experimental](../../../document-status.md)
+
+This document defines semantic conventions to apply when creating a span for requests to Elasticsearch.
+
+## Span Name
+
+The **span name** SHOULD be of the format `<http.request.method> <url.path *with placeholders*>`.
+
+The elasticsearch url path is used with placeholders in order to reduce the cardinality of the span name. When the
+path contains a document id, it SHOULD be represented as the identifier `{id}`. When the path contains a target data stream
+or index, it SHOULD be represented as `{target}`.
+For example, a request to `/test-index/_doc/123` should have the span name `GET /{target}/_doc/{id}`.
+When there is no target or document id, the span name contains the exact path, as in `POST /_search`.
+
+## Span attributes
+
+<!-- semconv db.elasticsearch -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `db.elasticsearch.doc_id` | string | The document that the request targets. | `123`; `456` | Conditionally Required: [1] |
+| `db.elasticsearch.target` | string | The name of the data stream or index that is targeted. | `users` | Conditionally Required: [2] |
```

Comment:
`{target}` seems to indicate a singular where it could be plural. Might be good to normalize `type` to `array` (strings).



Maybe we want to generalise this section to: 

`db.elasticsearch.parts.{namefromspec}` 
that way we can pick up url parts from any Elasticsearch call:

https://github.com/search?q=repo%3Aelastic%2Felasticsearch-specification%20path_parts&type=code


That would make this available as `db.elasticsearch.parts.index` as per 
https://github.com/elastic/elasticsearch-specification/blob/main/specification/_global/search/SearchRequest.ts#L60 but we could discuss renaming this part to `target` since for a long time it does not solely apply to indices anymore.

Reply:
@estolfo @Mpdreamz This should be resolved now with the changes in ad963747464a4f947e0751db92b16c5db6e0d692, right?

Reply:
Yup!